src/BigANNBinaryEmbeddingScenarioBase.cs

```diff
@@ -188,6 +196,60 @@ protected async Task BulkIngestDataForRange(IngestionOperationType ingestionOper
             }
         }
 
+        private async Task RetryFailedIds(ConcurrentBag<int> failedIds, IngestionOperationType ingestionOperationType, int? startTagId)
```

Comment:
can you also think this from streaming ingestion perspective? Hope this is not regressing that.